src/lib/y2users/users_simple/writer.rb

```diff
@@ -33,9 +33,18 @@ def initialize(config)
       end
 
       # Stores all users from {#config} into the UsersSimple module
+      #
+      # UsersSimple module is not expected to contain the root user in its list of users. It
+      # provides separate attributes in order to store the root user info (e.g., root_password or
+      # root_public_key).
       def write
-        users_simple = config.users.map { |u| to_user_simple(u) }
+        users_simple = config.users.reject(&:root?).map { |u| to_user_simple(u) }
         Yast::UsersSimple.SetUsers(users_simple)
+
+        root = config.users.find(&:root?)
+        return unless root
+
+        Yast::UsersSimple.SetRootPassword(root.password.value) unless root.password.nil?
```

Comment:
please add todo as I think here we set plain password and not encrypted one, right?